content.rb

```diff
@@ -0,0 +1,105 @@
+require 'google/apis/analyticsreporting_v4'
+require 'date'
+require 'json'
+
+class Content
+  attr_reader :service, :credentials
+  def initialize
+    @service = Google::Apis::AnalyticsreportingV4::AnalyticsReportingService.new
+    @credentials = Google::Auth::ServiceAccountCredentials.make_creds(
+      scope: "https://www.googleapis.com/auth/analytics.readonly"
+    )
+    @service.authorization = @credentials
+  end
+
+  # Return formatted page data for 10 most popular pages on GOV.UK
+  def most_popular_govuk_pages
+    format_response.to_json
+  end
+
+private
+
+  # Make request to Google Analytics Core Reporting API 
+  def ga_request
+    service.batch_get_reports(analytics_reports)
+  end
+
+  # Build Google Analytics report request
+  def analytics_reports
+    Google::Apis::AnalyticsreportingV4::GetReportsRequest.new(
+      { report_requests: [build] }
+    )
+  end
+
+  # Format response
+  def format_response
+    begin
+      ga_response = ga_request.reports.first.to_h
+    rescue Google::Apis::Error => e
+      puts "#{e.message}"
+      page_data = [
+        {
+          page_views: "No data available",
+          page_path: "",
+          page_title: "No data available"
+        }
+      ]
+    else
+      page_data = []
+        ga_response[:data][:rows].each do |row|
+          row_data = {
+            page_views: row[:metrics][0][:values][0],
+            page_path: row[:dimensions][0],
+            page_title: row[:dimensions][1]
+          }
+          page_data << row_data
+        end
+      page_data
+    end
+  end
+
+  # Build the GA query
+  def build
+    Google::Apis::AnalyticsreportingV4::ReportRequest.new(
+      view_id: 'ga:53872948',
+      sampling_level: 'LARGE',
+      date_ranges: [date_range],
+      metrics: [metric],
+      order_bys: [{field_name: 'ga:pageviews', sort_order: 'DESCENDING'}],
+      dimensions: [dimension_path, dimension_title],
+      page_size: '10'
+    )
+  end
+
+  # Set the date range
+  def date_range
+    start_date = Date.today - 6
+    end_date = Date.today
+
+    Google::Apis::AnalyticsreportingV4::DateRange.new(
+      start_date: start_date.to_s,
+      end_date: end_date.to_s
+    )
+  end
+
+  # Set the metric
```

Comment:
Similar suggestions to the other methods - could we rename and remove the need for the comments?